### ISSUE TITLE
More Details Needed for Reproduction

### ISSUE THREAD

**Issue by DarrenJum** [mamengyiyi/CPI#1]
I have carefully followed the settings for the regularization parameter and weighting coefficient as described in the paper. However, despite my best efforts, I am unable to achieve the same performance levels reported, for either the CPI or CPI_RE policy.

To help me reproduce the experiment, I would be very grateful if the auther could provide more details.

**Comment by mamengyiyi** [comment]
Hi, could  u please provide the reproduced results, the used hyperparameters and the corresponding dataset?

**Comment by DarrenJum** [comment]
I have made the following modifications to the code and would be grateful if you could help me identify any errors that might be causing discrepancies.

For the hopper-medium-v2 environment, I used the settings from Table 4 in the paper: regularization parameter qweight=0.2, weighting coefficient aweight=0.5, and normalize=True. All other parameters were kept at their default values in the code. The CPI policy was missing the "device" parameter, which I have corrected, and the code now runs. However, the final D4RL scores for both CPI and CPI_RE are consistently under 30.

My environment's main component versions are:
- python==3.8.5
- pytorch==1.10.0
- mujoco210
- mujoco-py== 2.1.2.14
- gym==0.18.3

**Comment by mamengyiyi** [comment]
Hi, I rerun the CPI code on hopper-medium-v2  using the following command:
`python main.py --env hopper-medium-v2 --qweight=0.2 --aweight=0.5 --normalize
`
And below are the partial running logs I screenshot during the training for 1000000 steps:

<img width="1256" alt="Image" src="https://github.com/user-attachments/assets/3d421ec2-8d53-4786-a0d4-7651c6045253" />

<img width="1268" alt="Image" src="https://github.com/user-attachments/assets/2327d02d-41f9-4a1a-87ab-b2ff0a1441bb" />

It seems the code works well. 

The rerun are performed on environment with following main components:

- GPU: NVIDIA 4090
- python==3.8.6
- pytorch==2.4.1 (when we publish the paper, the torch version is also 1.XX)
- mujoco210
- mujoco-py== 2.1.2.14

So I guess the main reason is normalization didn't work when you set normalize=True as normalize is a store_true action. Could u please try the command I provided and see what's going on?

**Comment by DarrenJum** [comment]
Thank you for your active and patient reply. I found that the previous error occurred because I had been testing the code on the CPI_RE policy, and coincidentally the parameters for CPI were not very suitable for hopper-medium-v2. I used tensorboard to print the score and Q value during the training process of CPI_RE using --qweight=0.2 --aweight=0.5 --normalize, and ran the code on other tasks. I found that only the Q values of hopper-medium-v2 showed accumulation of extrapolation error. I think at this time, qweight should be set slightly larger.

![Image](https://github.com/user-attachments/assets/e7b82b1b-1b62-4ad9-97b5-1775f1e608cc)

I will actively explore other parameter settings, and thank the author again for the active and patient response.

**Comment by mamengyiyi** [comment]
Thanks for your attention. Please feel free to contact me encontering any questions.
